### PR TITLE
Fixes #FXIOS-1519 - Added telemetry events for library and sync view

### PR DESCRIPTION
**Pull request by nbhasin2**
- Cleaned up un-used downloads event that was not being sent to glean
- Re-structured bookmarks panel event into library panel object

### CODE REVIEW

**Comment by nbhasin2** [comment]
# Request for data collection review form 
1. What questions will you answer with this data?
	* How many users access different libraray panel items like Bookmarks, History, Reading List, Downloads and Synced Tabs
	* How users access and use synced tabs and library panel 

2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:
	* This will help to compare results after we move synced tabs to the tabs tray and prioritize future bugs
	* It will also provide information essential for understanding importance of each library panel item and prioritize future work based on it

3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
	* No other way to collect this data

4. Can current instrumentation answer these questions?
	* No, we didn't have any events for Sync Library 
	* Events for library item were missing as well - History, Reading List, Downloads and Synced Tabs
	* We had an event for bookmarks view which is now moved to be a part of library panel item

5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.

Measurement Description | Data Collection Category | Tracking Bug # |
| --- | ----- | ----- |
Counts the number of times a specific library panel button is tapped for Bookmarks, History, Reading List, Downloads and Synced tabs | 2 | https://github.com/mozilla-mobile/firefox-ios/pull/8152
Counts the number of times a user taps on sign in to sync button in sync library view | 2 | https://github.com/mozilla-mobile/firefox-ios/pull/8152
Counts the number of times a user opens synced tab | 2 | https://github.com/mozilla-mobile/firefox-ios/pull/8152
Counts the number of times a user opens synced tab list from home view | 2 | https://github.com/mozilla-mobile/firefox-ios/pull/8152
Counts the number of times a user taps on create account button in sync library view | 2 | https://github.com/mozilla-mobile/firefox-ios/pull/8152


6. How long will this data be collected? Choose one of the following:
	* I want this data to be collected for 6 months initially (potentially renewable).

7. What populations will you measure?
All countries and channels that have data collection on.

8. If this data collection is default on, what is the opt-out mechanism for users?
	* Toggle switch for 'Send Usage Data' in the app settings.

9. Please provide a general description of how you will analyze this data.
	* By monitoring the count of how often users are performing these actions.

10. Where do you intend to share the results of your analysis?
	* With the mobile team.

11. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:
	* No.
	
Documentation has been updated in this PR and will be reflected in our [metrics.md](https://github.com/mozilla-mobile/firefox-ios/blob/main/Docs/metrics.md) file

File: Client/Telemetry/TelemetryWrapper.swift [hunk 1]
```diff
@@ -509,6 +517,18 @@ extension TelemetryWrapper {
             GleanMetrics.Widget.mTopSitesWidget.add()
         case (.action, .tap, .pocketStory, _, _):
             GleanMetrics.Pocket.openStory.add()
+        // Library
+        case (.action, .view, .libraryPanel, let type, _):
```
Comment (dnarcese) [inline Client/Telemetry/TelemetryWrapper.swift:521]: Should the method here be `.tap` instead of .view?

**Comment by dnarcese** [comment]
r+

# Data Review Form

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, telemetry is documented in `docs/metrics.md`

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, Firefox iOS provides data controls in the app settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

This should be re-reviewed w/ Ashley Thomas every 6mo

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Type 2

5) Is the data collection request for default-on or default-off?

Default on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does there need to be a check-in in the future to determine whether to renew the data? (Yes/No) (If yes, set a todo reminder or file a bug if appropriate)**

Should review all probes for necessity every 6mo

9) Does the data collection use a third-party collection tool? 

No

File: Client/Telemetry/TelemetryWrapper.swift (hunk 1 above)
Comment (nbhasin2) [reply Client/Telemetry/TelemetryWrapper.swift]: fixed in cleanup commits

**Review by dnarcese — APPROVED** [review]

**Comment by nbhasin2** [comment]
Tests passed before in my last cleanup commit but failed for `Cleanup x3`. Change in last commit has nothing to do with failing test so I'll run BR again and see if it passes. If not I'll merge this in as these are just intermittent test failures from BR :\

**Comment by nbhasin2** [comment]
Merging as a completely different test failed and tests actually passed for the following commit 
Commit: https://github.com/mozilla-mobile/firefox-ios/runs/2123153041

Other two commits are not related to the test that failed

**Pull request merged**